File: src/main/java/com/mikea/gae/rx/RxModule.java
package com.mikea.gae.rx;

import com.google.inject.servlet.ServletModule;

/**
 * @author mike.aizatsky@gmail.com
 */
public class RxModule extends ServletModule {
    public static final String RX_URL_BASE = "/_rx";
    public static final String RX_CRON_URL_BASE = RxModule.RX_URL_BASE + "/cron/";

    @Override
    protected void configureServlets() {
        super.configureServlets();
        serve(RX_URL_BASE + "*").with(RxServlet.class);
        bind(Rx.class).to(RxImpl.class);
    }
}

File: src/main/scala/com/mikea/gae/rx/RxModule.scala
package com.mikea.gae.rx

import com.google.inject.servlet.ServletModule
import com.googlecode.objectify.ObjectifyService
import com.mikea.gae.rx.model.AppVersion
import com.mikea.gae.rx.impl.{RxFilter, RxImpl}

class RxModule extends ServletModule {
  protected override def configureServlets(): Unit = {
    super.configureServlets()

    ObjectifyService.register(classOf[AppVersion])

    filter("*").through(classOf[RxFilter])
    bind(classOf[Rx]).to(classOf[RxImpl])
  }
}